src/spring-cloud/azext_spring_cloud/_validators_enterprise.py

```diff
@@ -36,6 +36,21 @@ def not_support_enterprise(cmd, namespace):
         raise ClientRequestError("'{}' doesn't support for Enterprise tier Spring instance.".format(namespace.command))
 
 
+def validate_build_env(cmd, namespace):
+    if namespace.resource_group and namespace.service and not is_enterprise_tier(cmd, namespace.resource_group, namespace.service) and namespace.build_env is not None:
+        raise ClientRequestError("'--build-env' only supports for Enterprise tier Spring instance.")
+
+
+def validate_target_module(cmd, namespace):
+    if namespace.resource_group and namespace.service and is_enterprise_tier(cmd, namespace.resource_group, namespace.service) and namespace.target_module is not None:
+        raise ClientRequestError("'--target-module' doesn't support for Enterprise tier Spring instance.")
+
+
+def validate_runtime_version(cmd, namespace):
+    if namespace.resource_group and namespace.service and is_enterprise_tier(cmd, namespace.resource_group, namespace.service) and namespace.runtime_version is not None:
+        raise ClientRequestError("'--runtime-version' doesn't support for Enterprise tier Spring instance.")
```

Comment:
Is error `ArgumentUsageError` more appropriate than error `ClientRequestError`?

Reply:
Sounds good. Updated.